aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -143,6 +155,24 @@ Tensor where(const Tensor& condition, const Tensor& self, const Tensor& other) {
   return at::_s_where(b_condition, b_self, b_other);
 }
 
+Tensor where(const Tensor& condition, Scalar self, const Tensor& other) {
+  return at::where(condition, wrapped_scalar_tensor(self, other.device()), other);
+}
+
+Tensor where(const Tensor& condition, const Tensor& self, Scalar other) {
+  return at::where(condition, self, wrapped_scalar_tensor(other, self.device()));
+}
+
+Tensor where(const Tensor& condition, Scalar self, Scalar other) {
+  TORCH_CHECK(self.type() == other.type(),
```

Comment:
Is this check redundant with the check in `where(tensor, tensor, tensor)`?

Reply:
(line 181 of the current file)

Reply:
Makes sense. Done.